velox/exec/fuzzer/WindowFuzzer.cpp

```diff
@@ -707,58 +701,105 @@ bool WindowFuzzer::verifyWindow(
+    if (customVerifier) {
+      customVerifier->reset();
+    }
+  };
+
   velox::fuzzer::ResultOrError resultOrError;
   try {
     resultOrError = execute(plan);
     if (resultOrError.exceptionPtr) {
       ++stats_.numFailed;
     }
 
-    if (!customVerification) {
-      if (resultOrError.result && enableWindowVerification) {
-        auto prestoQueryRunner =
-            dynamic_cast<PrestoQueryRunner*>(referenceQueryRunner_.get());
-        bool isPrestoQueryRunner = (prestoQueryRunner != nullptr);
-        if (isPrestoQueryRunner) {
-          prestoQueryRunner->queryRunnerContext()
-              ->windowFrames_[windowNodeId]
-              .push_back(prestoFrameClause);
-        }
-        auto referenceResult =
-            computeReferenceResults(plan, referenceQueryRunner_.get());
-        if (isPrestoQueryRunner) {
-          prestoQueryRunner->queryRunnerContext()->windowFrames_.clear();
+    if (resultOrError.result) {
+      if (!customVerification) {
+        if (enableWindowVerification) {
+          auto prestoQueryRunner =
+              dynamic_cast<PrestoQueryRunner*>(referenceQueryRunner_.get());
+          bool isPrestoQueryRunner = (prestoQueryRunner != nullptr);
+          if (isPrestoQueryRunner) {
+            prestoQueryRunner->queryRunnerContext()
+                ->windowFrames_[windowNodeId]
+                .push_back(prestoFrameClause);
+          }
+          auto referenceResult =
+              computeReferenceResults(plan, referenceQueryRunner_.get());
+          if (isPrestoQueryRunner) {
+            prestoQueryRunner->queryRunnerContext()->windowFrames_.clear();
+          }
+          stats_.updateReferenceQueryStats(referenceResult.second);
+          if (auto expectedResult = referenceResult.first) {
+            ++stats_.numVerified;
+            stats_.verifiedFunctionNames.insert(
+                retrieveWindowFunctionName(plan)[0]);
+            VELOX_CHECK(
+                assertEqualResults(
+                    expectedResult.value(),
+                    plan->outputType(),
+                    {resultOrError.result}),
+                "Velox and reference DB results don't match");
+            LOG(INFO) << "Verified results against reference DB";
+          }
         }
-        stats_.updateReferenceQueryStats(referenceResult.second);
-        if (auto expectedResult = referenceResult.first) {
-          ++stats_.numVerified;
-          stats_.verifiedFunctionNames.insert(
-              retrieveWindowFunctionName(plan)[0]);
-          VELOX_CHECK(
-              assertEqualResults(
-                  expectedResult.value(),
-                  plan->outputType(),
-                  {resultOrError.result}),
-              "Velox and reference DB results don't match");
-          LOG(INFO) << "Verified results against reference DB";
+      } else if (referenceQueryRunner_->supportsVeloxVectorResults()) {
+        if (enableWindowVerification) {
+          auto prestoQueryRunner =
```

Comment:
nit code under these if blocks is very similar, maybe we can move some of it to some common function ?